s3-handler/albevent/albevent.go

```diff
@@ -0,0 +1,188 @@
+package albevent
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/honeycombio/agentless-integrations-for-aws/common"
+	"github.com/honeycombio/honeytail/httime"
+	"github.com/honeycombio/libhoney-go"
+)
+
+type Event struct {
```

Comment:
I'm not sure we need to create a custom type, having a function that does most of what's in NewEvent and return a standard libhoney event would work just as well.

Reply:
@MikeGoldsmith part of this is _partly_ due to my misunderstanding that the entire intent of this handler was to process ALB logs from S3, but it _seems_ like it is even more general than that.

The other motivation was so that we could test _what_ values were actually being set on the event.

If this is made more general, what should happen to things like:

*  `addTraceId` (including the logic being done to convert an assumed `X-Amzn-Trace-Id` into a W3C compliant one)
*  `addUrlAttributes` which assumes that there is a `"request"` value and breaks out the useful parts of it
* `handleTimestamps` which assumes also ALB traffic so leveraging `target_processing_time` to derive `duration_ms`

Just curious if that should just be a feature flag or something to opt into that behavior / extract the common bits so that if you are _not_ processing ALB logs that you get the common non-ALB parts of that class or what not.